pkg/apis/build/v1alpha1/buildrun_types.go

```diff
@@ -176,7 +181,7 @@ type ServiceAccount struct {
 	Name *string `json:"name,omitempty"`
 	// If generates a new ServiceAccount for the build
 	// +optional
-	Generate bool `json:"generate,omitempty"`
+	Generate *bool `json:"generate,omitempty"`
```

Comment:
This similar, we here do not need a tri-state boolean (`nil`, `false`, `true`). If the field does not get set, it gets set to `false`, which is fine.